src/Microsoft.DotNet.Watcher.Core/Internal/Implementation/Project.cs

```diff
@@ -15,12 +17,27 @@ public Project(ProjectModel.Project runtimeProject)
             ProjectFile = runtimeProject.ProjectFilePath;
             ProjectDirectory = runtimeProject.ProjectDirectory;
 
-            Files = runtimeProject.Files.SourceFiles.Concat(
-                    runtimeProject.Files.ResourceFiles.Values.Concat(
-                    runtimeProject.Files.PreprocessSourceFiles.Concat(
-                    runtimeProject.Files.SharedFiles))).Concat(
-                    new string[] { runtimeProject.ProjectFilePath })
-                .ToList();
+            var compilerOptions = runtimeProject.GetCompilerOptions(targetFramework: null, configurationName: null);
+
+            var filesToWatch = new List<string>() { runtimeProject.ProjectFilePath };
+            if (compilerOptions.CompileInclude != null)
+            {
+                filesToWatch.AddRange(compilerOptions.CompileInclude.ResolveFiles());
+
+                if (compilerOptions.EmbedInclude != null)
```

Comment:
That's a good question, maybe you have an answer. So this new project model has two sets of properties: the old ones and the new ones. 

1. Can you have both of them set at the same time? 
1. Does having one set mean ignore the other one? 
1. Which one takes priority because I see that the new properties are `null` by default if not set but the old ones have default values if not set